javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -2702,23 +2701,20 @@ public static boolean isVarargsCall(MethodInvocationTree invok) {
             return true;
         }
 
-        // For some calls the varargsElement element disappears when it should not. This seems to
-        // only be a problem with MethodHandle#invoke and only with no arguments.  See
-        // framework/tests/all-systems/Issue6078.java.
-        // So also check for a mismatch between parameter and argument size.
-        // Such a mismatch occurs for every enum constructor: no args, two params (String name, int
-        // ordinal).
-
-        List<? extends VariableElement> parameters = elementFromUse(invok).getParameters();
-        int numParameters = parameters.size();
-        if (numParameters != invok.getArguments().size()) {
-            if (numParameters > 0
-                    && parameters.get(numParameters - 1).asType() instanceof ArrayType) {
-                return true;
-            }
-        }
+        return isSignaturePolymorphic(invok);
+    }
 
-        return false;
+    /**
+     * Returns true if the given method invocation targets a signature polymorphic method. See <a
```

Comment:
I would have also written `signature-polymorphic method`, but the JLS consistently writes it without dash, so let's go with that.